src/main/java/i18n/ResourceHandler.java

```diff
@@ -0,0 +1,108 @@
+package i18n;
+
+import java.text.DateFormat;
+import java.text.Format;
+import java.text.MessageFormat;
+import java.util.Locale;
+import java.util.ResourceBundle;
+
+/**
+ * Static class providing localized messages from the i18n message bundles.
+ */
+public class ResourceHandler {
+    
+    private static final String DEFAULT_MESSAGE_BUNDLE_PATH = "i18n/MessageBundle";
+    
+    protected static class ResourceHandlerInstance {
+        
+        protected ResourceHandlerInstance(String messageBundlePath) {
+            this.messageBundlePath = messageBundlePath;
+            
+            locale = Locale.getDefault();
+            
+            loadResourceBundle();
+        }
+        
+        private final String messageBundlePath;
+
+        private Locale locale;
+        private ResourceBundle resourceBundle;
+        
+        protected Locale getLocale() {
+            return locale;
+        }
+        
+        protected void setLocale(final Locale locale) {
+            this.locale = locale;
+            
+            loadResourceBundle();
+        }
+        
+        protected String getMessage(final String key, final Object... args) {
+            String message = resourceBundle.getString(key);
+            
+            MessageFormat format = new MessageFormat("");
+            format.setLocale(locale);
+            format.applyPattern(message);
+            replaceUnsupportedFormats(format);
+            
+            return format.format(args);
+        }
+        
+        private void loadResourceBundle() {
+            resourceBundle = ResourceBundle.getBundle(messageBundlePath, locale);
+        }
+        
+        private static void replaceUnsupportedFormats(MessageFormat format) {
+            Format[] subformats = format.getFormats();
+            
+            for (int i = 0; i < subformats.length; i++) {
+                if (subformats[i] instanceof DateFormat) {
+                    format.setFormat(i, new DateFormatWrapper((DateFormat) subformats[i]));
+                }
```

Comment:
I'm not the greatest supporter of these instanceof-styled solutions, but I don't have a cleaner solution in mind either. We should keep that in mind and maybe replace it at some point.

Reply:
https://github.com/kit-sdq/TimeSheetGenerator/pull/76#discussion_r409545499